### PR TITLE
feat: add typescript type definitions

### PR DESCRIPTION
**Pull request by shahbhavir**
> This PR will add typescript definitions, as well as one function (Matic.setWallet) required for typescript compatibility. Further explanation in commit description.
from https://github.com/maticnetwork/matic.js/pull/46

_We have made just one change to this PR by moving declaration file to `types` folder, so it is available to the developers using this library._

@dylanbienenstock Thank you once again for your efforts in improving developer experience while using `maticjs` library 👍

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+export type WeiAmount = string;
+export type TokenId = string;
+export type MaticChainTransactionId = string;
+
+/**
+ * * `from` must be valid account address(required)
+ * * `gasPrice` same as Ethereum sendTransaction
+ * * `gasLimit` same as Ethereum sendTransaction
+ * * `nonce` same as Ethereum sendTransaction
+ * * `nonce` same as Ethereum sendTransaction
```
Comment (ptsayli) [inline types/index.d.ts]: duplicate

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
+import { Account, Provider, Eth, Tx, TransactionReceipt, Transaction, PromiEvent } from 'web3/types';
+import Web3 from 'web3';
+
+// These types are declared to help you keep track of similar data
+export type UserMaticAddress = string;
```
Comment (ptsayli) [inline types/index.d.ts]: in my understanding, `UserMaticAddress` is `tokenContractAddress` on Matic
so maybe variable name can be something else

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+export default class Matic {
+    /**
+     * Create Matic SDK instance with the given options
+     * @param options 
+     */
+    constructor(options: MaticConstructor);
+    web3: Web3;
+    parentWeb3: Web3;
+    wallet: Account; 
+    walletAddress: UserMaticAddress;
```
Comment (ptsayli) [inline types/index.d.ts]: what is `walletAddress` ? 
`user address` to which PK belongs?

File: types/index.d.ts (hunk 2 above)
Comment (0xAshish) [reply types/index.d.ts]: `UserEthAddress` and `UserMaticAddress` are same so this can be `UserAddress`
@ptsayli @shahbhavir

Comment (0xAshish) [reply types/index.d.ts]: what is `EthContractAddress`?

**Comment by 0xAshish** [comment]
wait for @ptsayli's review

**Pull request merged**